File: leetcode/remove_duplication_I.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[k] = nums[i]
                k += 1

        return k # return the number of elements are not duplication
    
if __name__ == '__main__':
    s = Solution()
    nums = [0, 0, 0, 1, 1, 2, 4, 6, 7, 7]
    print(s.removeDuplicates(nums))

File: leetcode/remove_duplication_I.cpp
using namespace std;
#include <iostream>
#include <vector>


class Solution {
public: 
    int removeDuplicates(vector<int>&nums) {
        int k = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != nums[i-1]) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

int main() {
    Solution s;
    vector<int> nums = {1, 1, 1, 3, 4, 6, 6, 7, 7};
    cout << s.removeDuplicates(nums) << endl;
}